api/TiFAPI.ts

```diff
@@ -1,393 +1,41 @@
-    return await this.apiFetch(
-      {
-        method: "PATCH",
-        endpoint: `/user/block/${id}`
-      },
-      {
-        status204: "no-content",
-        status404: UserNotFoundResponseSchema
-      }
-    )
-  }
-
-  /**
-   * Unblocks a user.
-   *
-   * Blocks can be mutual (ie. 2 users can block each other regardless of if
-   * they have been blocked by the user they are trying to block), so this
-   * endpoint only fails if the given user id does not exist or if the user
-   * is not blocked in the first place.
-   */
-  async unblockUser(id: UserID) {
-    return await this.apiFetch(
-      {
-        method: "DELETE",
-        endpoint: `/user/block/${id}`
-      },
-      {
-        status204: "no-content",
-        status404: UserNotFoundResponseSchema,
-        status403: userTiFAPIErrorSchema("user-not-blocked")
-      }
-    )
-  }
-
-  /**
-   * Returns the events in the area of the center with the given radius in
-   * meters.
-   */
-  async exploreEvents(
-    center: LocationCoordinate2D,
-    radiusMeters: number,
-    signal?: AbortSignal
-  ) {
-    return await this.apiFetch(
-      {
-        method: "POST",
-        endpoint: "/event/region",
-        body: {
-          userLatitude: center.latitude,
-          userLongitude: center.longitude,
-          radius: radiusMeters
-        }
-      },
-      { status200: EventsInAreaResponseSchema },
-      signal
-    )
-  }
-
-  /**
-   * Returns the remote settings for the current user.
-   */
-  async userSettings() {
-    return await this.apiFetch(
-      {
-        method: "GET",
-        endpoint: "/user/self/settings"
-      },
-      { status200: UserSettingsResponseSchema }
-    )
-  }
-
-  /**
-   * Saves the specified user settings and returns the updated user settings.
-   */
-  async saveUserSettings(request: UpdateUserSettingsRequest) {
-    return await this.apiFetch(
-      {
-        method: "PATCH",
-        endpoint: "/user/self/settings",
-        body: request
-      },
-      { status200: UserSettingsResponseSchema }
-    )
-  }
-}
-
 /**
  * A high-level client for the TiF API.
  *
  * This class provides wrapper functions which use the lower level
- * {@link TiFAPITransport} function. Function parameters and return values in
+ * {@link tifAPIClient} function. Function parameters and return values in
  * this class represent honest data from the API, and not the exact data that
  * the UI may need. Therefore, this class should not be used directly in a UI
  * or frontend layer (eg. react), and should instead be used inside a data
  * layer that the UI can call into.
  */
-export const TiFAPI = _TiFAPIClass as TiFAPIConstructor
+export const TiFAPI = {
+  testAuthenticatedInstance: implementTiFAPI(
+    tifAPIClient(
+      TEST_API_URL,
+      jwtMiddleware(
+        async () => "I was here at the beginning, and I will proclaim the end."
+      )
+    )
+  ),
+  
+  testPath: (endpoint) => `${TEST_API_URL}${endpoint.slice(1)}`
```

Comment:
Is it possible that we can get some better msw utils that make testing code that uses the `TiFAPI` easier?

With this new "schema" approach it should be a lot easier as the endpoint path is centralized and not encapsulated in the body of a class method like how it was done previously.